homeassistant/components/surepetcare/__init__.py

```diff
@@ -0,0 +1,176 @@
+    },
+    extra=vol.ALLOW_EXTRA,
+)
+
+
+async def async_setup(hass, config):
+    """Initialize the Sure Petcare component."""
+    conf = config[DOMAIN]
+
+    # update interval
+    scan_interval = conf.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
+
+    # shared data
+    hass.data[DOMAIN] = hass.data[DATA_SURE_PETCARE] = {}
+
+    # sure petcare api connection
+    try:
+        surepy = SurePetcare(
+            conf[CONF_USERNAME],
+            conf[CONF_PASSWORD],
+            conf[CONF_HOUSEHOLD_ID],
+            hass.loop,
+            async_get_clientsession(hass),
+            debug=True,
+        )
+    except SurePetcareAuthenticationError:
+        _LOGGER.error("Unable to connect to surepetcare.io: Wrong credentials!")
+        return False
+    except (SurePetcareError, SurePetcareConnectionError) as error:
+        _LOGGER.error("Unable to connect to surepetcare.io: Wrong %s!", error)
+        return False
+
+    # add flaps
+    things = [
+        {
+            CONF_NAME: flap[CONF_NAME],
+            CONF_ID: flap[CONF_ID],
+            CONF_TYPE: SureThingID.FLAP.name,
+            CONF_DATA: await surepy.get_flap_data(flap[CONF_ID]),
+        }
+        for flap in conf[CONF_FLAPS]
+    ]
+
+    # add pets
+    things.extend(
+        [
+            {
+                CONF_NAME: pet[CONF_NAME],
+                CONF_ID: pet[CONF_ID],
+                CONF_TYPE: SureThingID.PET.name,
+                CONF_DATA: await surepy.get_pet_data(pet[CONF_ID]),
+            }
+            for pet in conf[CONF_PETS]
+        ]
+    )
+
+    spc_api = hass.data[DATA_SURE_PETCARE][SPC] = SurePetcareAPI(
+        hass, surepy, things, conf[CONF_HOUSEHOLD_ID]
+    )
+
+    # initial update
+    await spc_api.async_update()
+
+    async_track_time_interval(hass, spc_api.async_update, scan_interval)
+
+    # load platforms
+    hass.async_create_task(
+        hass.helpers.discovery.async_load_platform("binary_sensor", DOMAIN, {}, config)
+    )
+    hass.async_create_task(
+        hass.helpers.discovery.async_load_platform("sensor", DOMAIN, {}, config)
+    )
+
+    return True
+
+
+class SurePetcareAPI:
+    """Define a generic Sure Petcare object."""
+
+    def __init__(self, hass, surepy, ids, household_id):
+        """Initialize the Sure Petcare object."""
+        self.hass = hass
+        self.surepy = surepy
+        self.household_id = household_id
+        self.ids = ids
+        self.states = {}
+
+    async def async_update(self, args=None):
+        """Refresh Sure Petcare data."""
+        for thing in self.ids:
+            sure_id = thing[CONF_ID]
+            sure_type = thing[CONF_TYPE]
+
+            try:
+                if sure_type == SureThingID.FLAP.name:
+                    if SureThingID.FLAP.name not in self.hass.data[DATA_SURE_PETCARE]:
+                        self.hass.data[DATA_SURE_PETCARE][SureThingID.FLAP.name] = {}
+
+                    self.hass.data[DATA_SURE_PETCARE][SureThingID.FLAP.name][
```

Comment:
Don't save the data in hass.data. Save it in the `states` instance attribute that we've defined.